anesthetic/plot.py

```diff
@@ -281,6 +308,35 @@ def fastkde_plot_1d(ax, data, *args, **kwargs):
 
     ans = ax.plot(x[i], p[i], color=color, *args, **kwargs)
     ax.set_xlim(*check_bounds(x[i], xmin, xmax), auto=True)
+
+    if fill is True:
+        lims = iso_probability_1d(x[i], p[i], levels)
+        c = [round(1-level, 2) for level in levels]
+        for j in range(len(lims)):
+            if j < len(lims) - 1:
+                if lims[j] != lims[j+1]:
+                    x_per = np.linspace(lims[j], lims[j+1], 1000)
+                    y_per = np.interp(x_per, x[i], p[i])
```

Comment:
Why do we need an interpolation? Would x[i] and p[i] not be sufficient to provide the points (i.e. if you changed the above two lines for `x_per = x[i]`, `y_per=y[i]`, would anything change?

Reply:
y_per and x_per get used in the ax.fill_between function, might be able to simplify this but I will need to have a think.

Reply:
Have you tried just dropping x_per and y_per and replacing them with x[i] and y[i]?

Reply:
Yes this wont work unfortunately, you just get one filled region with the darkest shade of the colour map. x_per and y_per denote the sections of the pdf divided up at the limits. 

The figure below shows the different regions for the example distribution I am using here (each coloured line is a y_per-x_per). The limits of x_per denote the boundaries of each shaded region and the max value of y_per relative to the 'contour' levels gives the shade.

![sections](https://user-images.githubusercontent.com/40355093/107273769-6aaee080-6a47-11eb-9f05-1a1c35cecbb8.png)

Reply:
I think if we can replace the turning point based strategy with a CDF based one, then we should get the crossing points without the need for this interpolation, so you should focus on the CDF based issues